--- a/toolkit/components/telemetry/core/Telemetry.cpp
+++ b/toolkit/components/telemetry/core/Telemetry.cpp
@@ -1650,6 +1650,11 @@
 }

 nsresult TelemetryImpl::EarlyInit() {
+  // BOBZILLA: Hard-disable Telemetry Engine
+  // Instantly return an error condition to prevent any initialization of the
+  // internal Mozilla telemetry core, disregarding any preference toggles.
+  return NS_ERROR_NOT_INITIALIZED;
+
   MOZ_ASSERT(NS_IsMainThread());

   if (mInitialized) {
     return NS_OK;
